File: app/src/main/java/ru/nikshlykov/englishwordsapp/ui/word/WordDialogsViewModel.java
package ru.nikshlykov.englishwordsapp.ui.word;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;

import javax.inject.Inject;


import ru.nikshlykov.englishwordsapp.db.GroupsRepository;
import ru.nikshlykov.englishwordsapp.db.link.Link;

public class WordDialogsViewModel extends AndroidViewModel {

    private long wordId;

    private GroupsRepository groupsRepository;

    @Inject
    public WordDialogsViewModel(@NonNull Application application, GroupsRepository groupsRepository) {
        super(application);
        this.groupsRepository = groupsRepository;
    }

    public void setWordId(long wordId) {
        this.wordId = wordId;
    }

    // Методы для обработки результата работы диалога.
    public void deleteLink(long subgroupId) {
        Link linkToDelete = new Link(subgroupId, wordId);
        groupsRepository.delete(linkToDelete);
    }
    public void insertLink(long subgroupId) {
        Link linkToInsert = new Link(subgroupId, wordId);
        groupsRepository.insert(linkToInsert);
    }
}

File: app/src/main/java/ru/nikshlykov/englishwordsapp/ui/viewmodels/WordDialogsViewModel.kt
package ru.nikshlykov.englishwordsapp.ui.viewmodels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import ru.nikshlykov.englishwordsapp.db.GroupsRepository
import ru.nikshlykov.englishwordsapp.db.link.Link
import ru.nikshlykov.englishwordsapp.domain.interactors.AddWordToSubgroupInteractor

class WordDialogsViewModel(
  application: Application,
  private val groupsRepository: GroupsRepository,
  private val addWordToSubgroupInteractor: AddWordToSubgroupInteractor
) : AndroidViewModel(application) {
  private var wordId: Long = 0
  fun setWordId(wordId: Long) {
    this.wordId = wordId
  }

  // Методы для обработки результата работы диалога.
  fun deleteLink(subgroupId: Long) {
    val linkToDelete = Link(subgroupId, wordId)
    groupsRepository.delete(linkToDelete)
  }

  fun addWordToSubgroup(subgroupId: Long) {
    // TODO Поменять потом GlobalScope на свой.
    GlobalScope.launch {
      addWordToSubgroupInteractor.addLinkBetweenWordAndSubgroup(wordId, subgroupId)
    }
  }
}